sdk/storage/azure-storage-file-datalake/src/test/java/com/azure/storage/file/datalake/FileAsyncApiTests.java

```diff
@@ -424,12 +433,3792 @@ public void createOptionsWithTimeToExpireRelativeToNow() {
+        nullValues="null")
+    public void createIfNotExistsOptionsWithPathHttpHeaders(String cacheControl, String contentDisposition,String contentEncoding,
+                                                          String contentLanguage, byte[] contentMD5,String contentType) {
+        fc = dataLakeFileSystemAsyncClient.getFileAsyncClient(generatePathName());
+
+        PathHttpHeaders putHeaders = new PathHttpHeaders()
+            .setCacheControl(cacheControl)
+            .setContentDisposition(contentDisposition)
+            .setContentEncoding(contentEncoding)
+            .setContentLanguage(contentLanguage)
+            .setContentMd5(contentMD5)
+            .setContentType(contentType);
+
+        DataLakePathCreateOptions options = new DataLakePathCreateOptions().setPathHttpHeaders(putHeaders);
+
+        assertAsyncResponseStatusCode(fc.createIfNotExistsWithResponse(options, null), 201);
+    }
+
+    @ParameterizedTest
+    @CsvSource(value={"null,null,null,null","foo,bar,fizz,buzz"},nullValues="null")
+    public void createIfNotExistsOptionsWithMetadata(String key1, String value1, String key2, String value2) {
+        fc = dataLakeFileSystemAsyncClient.getFileAsyncClient(generatePathName());
+
+        Map<String, String> metadata = new HashMap<>();
+        if (key1 != null && value1 != null) {
+            metadata.put(key1, value1);
+        }
+        if (key2 != null && value2 != null) {
+            metadata.put(key2, value2);
+        }
+        DataLakePathCreateOptions options = new DataLakePathCreateOptions().setMetadata(metadata);
+
+        assertAsyncResponseStatusCode(fc.createIfNotExistsWithResponse(options, null), 201);
+
+        StepVerifier.create(fc.getProperties())
+            .assertNext(r -> {
+                for(String k : metadata.keySet()){
+                    assertTrue(r.getMetadata().containsKey(k));
+                    assertEquals(metadata.get(k), r.getMetadata().get(k));
+                }
+            })
+            .verifyComplete();
+    }
+
+    @Test
+    public void createIfNotExistsOptionsWithPermissionsAndUmask() {
+        fc = dataLakeFileSystemAsyncClient.getFileAsyncClient(generatePathName());
+        DataLakePathCreateOptions options = new DataLakePathCreateOptions().setPermissions("0777").setUmask("0057");
+        fc.createIfNotExistsWithResponse(options, null).block();
+
+        StepVerifier.create(fc.getAccessControlWithResponse(
+            true, null, null))
+            .assertNext(r -> assertEquals(PathPermissions.parseSymbolic("rwx-w----").toString(),
+                            r.getValue().getPermissions().toString()))
+            .verifyComplete();
+    }
+
+    @DisabledIf("olderThan20201206ServiceVersion")
+    @Test
+    public void createIfNotExistsOptionsWithLeaseId() {
+        fc=dataLakeFileSystemAsyncClient.getFileAsyncClient(generatePathName());
+
+        String leaseId=CoreUtils.randomUuid().toString();
```

Comment:
I would recommend using `testResourceNamer.randomUuid()` here so the test recording captures the UUID